cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -77,6 +83,85 @@ def _validate_device_specification(proto: v2.device_pb2.DeviceSpecification) ->
                     )
 
 
+def _build_gateset_and_gate_durations(
+    proto: v2.device_pb2.DeviceSpecification,
+) -> Tuple[cirq.Gateset, Dict[cirq.GateFamily, cirq.Duration]]:
```

Comment:
This is a private method, but I would still add a docstring since it's a little long.